hydro_lang/src/stream.rs

```diff
@@ -1573,6 +1574,112 @@ impl<'a, T, L: Location<'a> + NoTick + NoAtomic, B, Order> Stream<T, L, B, Order
         Stream::new(Atomic { tick: tick.clone() }, self.ir_node.into_inner())
     }
 
+    /// Consumes a stream of `Future<T>`, produces a new stream of the resulting `T` outputs.
+    /// Future outputs are produced as available, regardless of input arrival order.
+    ///
+    /// # Example
+    /// ```rust
+    /// # use std::collections::HashSet;
+    /// # use dfir_rs::futures::StreamExt;
+    /// # use hydro_lang::*;
+    /// # tokio_test::block_on(test_util::stream_transform_test(
+    ///     |process| {
```

Comment:
Ah, like the other doctests, you should move this line above, and also `#` out the assertion and only display a comment instead.